### PR TITLE
feat: add POST /connectors

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

1. add POST /connectors
2. add UTs.
3. add ITs.

## Testing

UTs and ITs.

### CODE REVIEW

File: packages/core/src/routes/connector.ts [hunk 1]
```diff
@@ -59,6 +67,57 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
     }
   );
 
+  router.post(
+    '/connectors/:connectorId',
+    koaGuard({
+      params: object({ connectorId: string() }),
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:73]: this should be placed in `body`

File: packages/core/src/routes/connector.ts [hunk 2]
```diff
@@ -59,6 +67,57 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+      }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { connectorId },
+        body: { config, metadata },
+        body,
+      } = ctx.guard;
+
+      const logtoConnectors = await loadConnectors();
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:88]: `standardConnectors`

File: packages/core/src/routes/connector.ts [hunk 3]
```diff
@@ -59,6 +67,57 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+      const standardConnector = logtoConnectors.find(({ metadata: { id } }) => id === connectorId);
+
+      if (!standardConnector) {
+        throw new RequestError({
+          code: 'entity.not_found',
+          status: 404,
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:94]: 422

File: packages/core/src/routes/connector.ts [hunk 4]
```diff
@@ -59,6 +67,57 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+        });
+      }
+
+      const { count } = await countConnectorByConnectorId(connectorId);
+      console.log(count, standardConnector.metadata.isStandard);
+      assertThat(
+        count === 0 || standardConnector.metadata.isStandard === true,
+        'connector.multiple_instances_not_supported'
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:102]: 422

File: packages/core/src/routes/connector.ts [hunk 5]
```diff
@@ -59,6 +67,57 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+      );
+
+      assertThat(!config || !metadata, 'connector.config_and_metadata_empty');
+
+      if (config) {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:107]: if `config` exist and pass the validation, this connector should be "enabled"?

Comment (darcyYe) [reply packages/core/src/routes/connector.ts:107]: This logic aligns to PATCH /connectors/:id, which validates the `config` and still needs to manually enable the connector.

File: packages/core/src/__mocks__/connector.ts [hunk 6]
```diff
@@ -3,7 +3,7 @@ import type { Connector } from '@logto/schemas';
 import { ConnectorType } from '@logto/schemas';
 import { any } from 'zod';
 
-import type { LogtoConnector } from '@/connectors/types';
+import type { LoadConnector, LogtoConnector } from '@/connectors/types';
```
Comment (wangsijie) [inline packages/core/src/__mocks__/connector.ts:6]: Need to align naming

File: packages/core/src/routes/connector.ts [hunk 7]
```diff
@@ -59,6 +67,57 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
...
+        count === 0 || standardConnector.metadata.isStandard === true,
+        new RequestError({
+          code: 'connector.multiple_instances_not_supported',
+          status: 422,
+        })
+      );
+
+      assertThat(!config || !metadata, 'connector.config_and_metadata_empty');
+
+      if (config) {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.ts:108]: I think we can remove "config" in this API endpoint for now. "metadata" is enough to create a "old version connector".

**Pull request closed without merging**